app/main.py

```diff
@@ -1,213 +1,53 @@
-            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
-
-    # arthur
-    arthur = knightsConfig["arthur"]
-
-    # apply armour
-    arthur["protection"] = 0
-    for a in arthur["armour"]:
-        arthur["protection"] += a["protection"]
-
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
-
+    knights = []
+
+    for knight_info in knights_config.values():
+        knight = Knight(knight_info)
+
+        for armour in knight.armour:
+            knight.apply_armour(Armour(armour))
+        knight.apply_weapon(Weapon(knight.weapon))
+        if knight.potion:
+            knight.apply_potion(Potion(knight.potion))
+
+        knights.append(knight)
+
+    lancelot, arthur, mordred, red_knight = None, None, None, None
+
+    for knight in knights:
+        if knight.name == "Lancelot":
+            lancelot = knight
+        elif knight.name == "Arthur":
+            arthur = knight
+        elif knight.name == "Mordred":
+            mordred = knight
+        elif knight.name == "Red Knight":
+            red_knight = knight
```

Comment:
```suggestion
    knights = {}

    for knight_info in knights_config.values():
        knight = Knight(knight_info)

        for armour in knight.armour:
            knight.apply_armour(Armour(armour))
        knight.apply_weapon(Weapon(knight.weapon))
        if knight.potion:
            knight.apply_potion(Potion(knight.potion))

        knights[knight.name] = knight
      
      lancelot = knights_dict.get("Lancelot")
      arthur = knights_dict.get("Arthur")
      mordred = knights_dict.get("Mordred")
      red_knight = knights_dict.get("Red Knight")
```

Reply:
let's remove the redundant code, since we only use one loop and the readability of the code will be better